src/ui/components/InputModal/InputModal.tsx

```diff
@@ -0,0 +1,182 @@
+    return (
+      (authentication.loggedIn &&
+        (authentication.userName === undefined ||
+          authentication.userName?.length === 0) &&
+        currentRoute?.path?.includes(TabsRoutePath.ROOT)) ||
+      !!missingAliasUrl
+    );
+  }, [
+    authentication.loggedIn,
+    authentication.userName,
+    currentRoute?.path,
+    missingAliasUrl,
+  ]);
+
+  const resolveConnectionOobi = async (content: string) => {
+    // Adding a pending connection item to the UI.
+    // This will be removed when the create connection process ends.
+    const connectionName = new URL(content).searchParams.get("name");
+
+    const pendingId = uuidv4();
+    dispatch(
+      updateOrAddConnectionCache({
+        id: pendingId,
+        label: connectionName || pendingId,
+        status: ConnectionStatus.PENDING,
+        connectionDate: new Date().toString(),
+      })
+    );
+
+    try {
+      await Agent.agent.connections.connectByOobiUrl(content);
+    } catch (e) {
+      const errorMessage = (e as Error).message;
+
+      const urlId = errorMessage
+        .replace(StorageMessage.RECORD_DOES_NOT_EXIST_ERROR_MSG, "")
+        .trim();
+
+      if (!urlId) {
+        showError("Scanner Error:", e, dispatch, ToastMsgType.SCANNER_ERROR);
+        return;
+      }
+
+      showError(
+        "Scanner Error:",
+        e,
+        dispatch,
+        ToastMsgType.DUPLICATE_CONNECTION
+      );
+
+      dispatch(setOpenConnectionId(urlId));
+    } finally {
+      dispatch(removeConnectionCache(pendingId));
+    }
+  };
+
+  const setUserName = () => {
+    Agent.agent.basicStorage
+      .createOrUpdateBasicRecord(
+        new BasicRecord({
+          id: MiscRecordId.USER_NAME,
+          content: {
+            userName: inputValue,
+          },
+        })
+      )
+      .then(() => {
+        dispatch(
+          setAuthentication({
+            ...authentication,
+            userName: inputValue,
+          })
+        );
+
+        setInputValue("");
+        dispatch(setToastMsg(ToastMsgType.USERNAME_CREATION_SUCCESS));
+      })
+      .catch((error) => {
+        showError(
+          "Unable to create user name",
+          error,
+          dispatch,
+          ToastMsgType.USERNAME_CREATION_ERROR
+        );
+      });
+  };
+
+  const handleConfirm = () => {
+    if (missingAliasUrl) {
+      if (!missingAliasUrl) return;
+      const url = new URL(missingAliasUrl);
+      url.searchParams.set("name", inputValue);
+      resolveConnectionOobi(url.toString());
+      dispatch(setMissingAliasUrl(undefined));
+      setInputValue("");
+      return;
+    }
+
+    setUserName();
+  };
+
+  const title = useMemo(() => {
+    if (missingAliasUrl) {
+      return i18n.t("inputmodal.title.connectionalias");
+    }
+
+    return i18n.t("inputmodal.title.username");
+  }, [missingAliasUrl]);
+
+  return (
+    <IonModal
+      isOpen={showModal}
+      id={componentId}
+      data-testid={`${componentId}-modal`}
```

Comment:
If the component name is "input-modal" that means the data-testid will be "input-modal-modal"? 😂 

Maybe remove the last modal from here.

Reply:
Or we could actually leave that and choose a different name for the component. InputModal seems not descriptive enough. How about InputRequest? Something that tells us this input cannot be skipped. What do you think?